src/rt/typeinfo/ti_ulong.d

```diff
@@ -27,7 +27,10 @@ class TypeInfo_m : TypeInfo
 
     override size_t getHash(scope const void* p)
     {
-        return hashOf(*cast(ulong*)p);
+        static if (ulong.sizeof <= size_t.sizeof)
+            return *cast(const ulong*)p;
+        else
+            return hashOf(*cast(const ulong*)p);
```

Comment:
Could this kind of logic be encapsulated in `hashOf` itself?

Reply:
It's not a big deal, but I think that we'll end up with instances of `hashOf` for most (all?) basic types anywa, so I'm unsure if this optimization is worth doing.